pdp-server/src/api/authz/forward_to_opa.rs

```diff
@@ -0,0 +1,177 @@
+use crate::errors::ApiError;
+use crate::state::AppState;
+use axum::http::StatusCode;
+use http::header::InvalidHeaderValue;
+use log::debug;
+use reqwest;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use serde_json::Value::{Bool, Object};
+use thiserror::Error;
+
+pub(crate) async fn send_request_to_opa<R: DeserializeOwned, B: Serialize>(
+    state: &AppState,
+    endpoint: &str,
+    body: &B,
+) -> Result<R, ForwardingError> {
+    // Create a new OPA request
+    let request = create_opa_request(body, state.settings.debug)?;
+
+    // Send the request to OPA
+    let response: OpaResponse<R> = send_raw_request_to_opa(state, endpoint, &request).await?;
+    Ok(response.result)
+}
+
+/// Generic function to forward requests to OPA
+async fn send_raw_request_to_opa<B: Serialize, R: DeserializeOwned>(
+    state: &AppState,
+    endpoint: &str,
+    body: &OpaRequest<B>,
+) -> Result<OpaResponse<R>, ForwardingError> {
+    // Build the OPA URL from the settings
+    let endpoint = endpoint.strip_prefix("/").unwrap_or(endpoint);
+    let opa_url = format!("{}/{}", state.settings.opa_url, endpoint);
+    debug!("Forwarding request to OPA at: {}", opa_url);
+
+    // Send request to OPA using the specialized OPA client
+    // TODO add timeout
```

Comment:
This should use the opa client query timeout for now, otherwise it is a breaking change for the customers expecting it to time out after that time